### PR TITLE
[LLM Runtime] Compare cpp logits with pytorch

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

Compare logits with pytorch logits for monitoring accuracy:

- [x] Introducing the output comparison method: `cmpData`
- [ ] fp32 output of cpp graph should be very close to that of pytorch: diff < 0.1%

cmpData:
`diff = sqrt(sum((out - ref) ^ 2) / sum((ref) ^ 2))`

## Expected Behavior & Potential Risk

diff of llama2 vs pytorch fp32
- fp32            0.00067020295
- q4_0            0.17200278
- q4_j_32       0.271983
- q8_j_bf16    1 (error  outputs)

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp [hunk 1]
```diff
@@ -72,6 +75,29 @@ class Model {
     generate_count = 0;
   }
 
+  py::array_t<float> forward(const std::vector<std::vector<model_token>>& input_ids) {
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp:78]: can we return vector directly to avoid introducing numpy dependency?

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp:78]: `#include <pybind11/numpy.h>`
- numpy is in pybind11, not including new dependency.
- which is also used by enabling gptq models.
https://github.com/intel/intel-extension-for-transformers/pull/611/files

File: tests/test_llm_runtime.py [hunk 2]
```diff
@@ -17,53 +18,76 @@ def setUpClass(cls):
...
+    #                               max_new_tokens=128, min_new_tokens=30, early_stopping=True,
+    #                               pad_token=pad_token)
+    #     # for i in range(len(itrex_generate_ids)):
+    #     #     self.assertListEqual(pt_generate_ids[i], itrex_generate_ids[i])
+
+
+    def test_llm_runtime(self):
+        # /mnt/disk1/data2/zhenweil/models/opt/opt-125m
+        model_name = "/mnt/disk1/data2/zhenweil/models/llama/Llama-2-7b-chat-hf"  # or local path to model
+        woq_config = WeightOnlyQuantConfig(compute_dtype="bf16", weight_dtype="int8", use_cache=True, use_ggml=False)
```
Comment (kevinintel) [inline tests/test_llm_runtime.py:77]: add int8-int4 test

**Comment by zhenwei-intel** [comment]
cherry-pick to https://github.com/intel/intel-extension-for-transformers/pull/685
